shop/src/main/java/ro/msg/learning/shop/service/SupplierService.java

```diff
@@ -0,0 +1,44 @@
+package ro.msg.learning.shop.service;
+
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.repository.exception.SupplierNotFoundException;
+import ro.msg.learning.shop.model.Supplier;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.repository.SupplierRepository;
+
+import java.util.List;
+
+@Service
+public class SupplierService {
+    private final SupplierRepository supplierRepository;
+
+    public SupplierService(SupplierRepository supplierRepository) {
+        this.supplierRepository = supplierRepository;
+    }
+
+    List<Supplier> all() {
+        return supplierRepository.findAll();
+    }
+
+    Supplier one(@PathVariable int id) {
+        return supplierRepository.findById(id).orElseThrow(() -> new SupplierNotFoundException(id));
+    }
+
+    Supplier supplier(@RequestBody Supplier supplier) {
+        return supplierRepository.save(supplier);
+    }
+
+    Supplier replaceSupplier(@RequestBody Supplier newSupplier, @PathVariable int id) {
```

Comment:
The service layer is used to contain the business logic and to facilitate the communication between the controller and the persistence layer.
Remove the **@PathVariable** and **@Requestbody** annotations from the parameters of your methods as they are also applicable only to methods of Spring controllers.
This applies also to the other services where you use them and also the request mappings.